src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -226,9 +226,10 @@ def handle_results(self, snuba_results, requested_query, projects):
                 ]
 
             for result in snuba_results['data']:
-                result['project.name'] = projects[result['project.id']]
-                if 'project.id' not in requested_query['selected_columns']:
-                    del result['project.id']
+                if 'project.id' in result:
```

Comment:
It's now possible this doesn't not exist in the result for time buckets that do not have any data.